### PR TITLE
This code orders articles by date descending, and handles invalid pat…

### PR DESCRIPTION
**Pull request by adamjamo**
…h errors

### CODE REVIEW

**Review by douglashellowell — APPROVED** [review]
Your files and functions are getting a bit messy

`fetchArticles` and `fetchArticlesById` are in `topics_models.js` try to keep things organised and labelled correctly!

Your tests are all mixed up as well - it would be a good idea to group the tests for endpoints together

---

You've got `comment_count` in your `fetchArticles` model but you aren't testing for it in your tests! make sure you add the tests before you add the features

---

you've got multiple "wrong-path" `404` tests - you only need one!

```
"/api/topiiicccaaaanna"
"/api/artivld"
```

---


much like when using `forEach` when testing an array you should always have an assertion to check the length of the array is greater than 1 when using `toBeSorted` - an empty array would always pass! 😱

---

remember to remove `console.logs` before commiting

---

Add the `comment_count` to your tests and have a bit of a tidy up then feel free to merge!

File: __tests__/app.test.js [hunk 1]
```diff
@@ -8,193 +8,245 @@ beforeEach(() => seed(data));
...
+    test("200 message: returns topics objects in an array", () => {
       return request(app)
-        .get("/api/topiiicccaaaanna")
-        .expect(404)
-        .then(({ body: { message } }) => {
-          expect(message).toBe("Path not found");
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body: { topics } }) => {
+          topics.forEach((topic) => {
```
Comment (P-Copley) [inline __tests__/app.test.js:19]: Need a length check on this one as well please

File: __tests__/app.test.js [hunk 2]
```diff
@@ -8,193 +8,245 @@ beforeEach(() => seed(data));
...
-          comment_count: "2",
+describe("GET /api/articles", () => {
+  describe("Happy Paths", () => {
+    test("Responds with an object containing a key of articles with array of articles", () => {
+      return request(app)
+        .get(`/api/articles`)
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toHaveLength(12);
+          console.log(body.articles, "check this");
```
Comment (P-Copley) [inline __tests__/app.test.js]:
```js
          console.log(body.articles, "check this");
```
This log snuck it's way in

File: __tests__/app.test.js [hunk 3]
```diff
@@ -8,193 +8,245 @@ beforeEach(() => seed(data));
...
+              created_At: expect.any(Number),
+              votes: expect.any(Number),
+            });
+          });
         });
+    });
+    describe("Error Handling", () => {
+      test("404 message: Returns 404 error if path is invalid", () => {
+        return request(app)
+          .get("/api/notauser")
```
Comment (P-Copley) [inline __tests__/app.test.js]: This is the same test you did above. No need to do it twice

File: __tests__/app.test.js [hunk 4]
```diff
@@ -8,193 +8,245 @@ beforeEach(() => seed(data));
...
+            comment_count: "2",
+          });
         });
     });
-    test("400 message: Returns 400 error if data type is incorrect", () => {
-      const article_id = "abc";
+    test("Responds with article including comment count", () => {
+      const article_id = 5;
       return request(app)
         .get(`/api/articles/${article_id}`)
```
Comment (P-Copley) [inline __tests__/app.test.js:86]: You've already testing this in the previous test. Consider testing an article that has 0 comments instead to make sure your JOIN works ok

**Pull request merged**